### PR TITLE
checks: Check for permissions not granted on macOS

### PR DESCRIPTION
**Pull request by prgmitchell**
### Description

Check for any permissions that are not granted on macOS and list them with a warning flag.
image

### Motivation and Context

Suggested on the community Discord server.

### How Has This Been Tested?

Tested logs with all permissions granted and with some missing.
https://obsproject.com/logs/cJpEIqQu07UoAdX1
https://obsproject.com/logs/rOK9qz0wdlw0KOQ3

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: checks/macos.py [hunk 1]
```diff
@@ -66,3 +66,28 @@ def checkRosettaTranslationStatus(lines):
...
+    for line in macPerms:
+        if 'denied' in line:
+            found = re.search(r'Permission for (.+) denied', line)
+            if found:
+                permissionName = found.group(1).title()
+                permissionDescription = {
+                    "Audio Device Access": "<b>Microphone</b>: OBS requires this permission if you want to capture your microphone.",
+                    "Video Device Access": "<b>Camera</b>: This permission is needed in order to capture content from a webcam or capture card.",
+                    "Accessibility": "<b>Accessibility</b>: For keyboard shortcuts (hotkeys) to work while other apps are focused this permission is needed.",
+                    "Screen Capture": "<b>Screen Recording</b>: OBS requires this permission to be able to capture your screen."
```
Comment (RytoEX) [inline checks/macos.py:84]: Keep the wording consistent. Use active voice.
```
OBS requires this permission [for X feature | to do X action]
```
or
```
OBS requires [Permission Name] [...]
```

File: checks/macos.py [hunk 2]
```diff
@@ -66,3 +66,28 @@ def checkRosettaTranslationStatus(lines):
...
+                }.get(permissionName)
+
+                if permissionDescription:
+                    deniedPermissions.append(permissionDescription)
+
+    if deniedPermissions:
+        deniedPermissionsString = "<br><ul><li>" + "</li><li>".join(deniedPermissions) + "</li></ul>"
```
Comment (RytoEX) [inline checks/macos.py:91]: Add line breaks appropriately or the console output becomes unreadable.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Requesting wording changes.

File: checks/macos.py (hunk 1 above)
Comment (prgmitchell) [reply checks/macos.py:84]: I can still adjust if you'd like but this language was taken directly from what is already written in OBS.

Screenshot 2024-03-14 at 11 12 35 AM

Comment (gxalpha) [reply checks/macos.py:84]: I believe this wording might come from me (although I may have been advised by others?), if there are better suggestions we can change it in OBS itself as well.

Comment (RytoEX) [reply checks/macos.py:84]: I did check the blame and you originally had all the strings consistently worded, but changed them based on feedback and I did not review that PR.

Comment (RytoEX) [reply checks/macos.py:84]: For the sake of getting this check applied to the analyzer, we can keep the strings consistent with OBS for now. I would like to revisit the strings in OBS in the future, and then also revisit them here.

File: checks/macos.py (hunk 2 above)
Comment (RytoEX) [reply checks/macos.py:91]: Please apply newlines as needed. We'll figure out how to get the output formatting (textwrap.fill) to behave another time, but at least the strings will be ready.

Comment (prgmitchell) [reply checks/macos.py]: Updated with new lines added.

**Comment by prgmitchell** [comment]
Resolved changes requested

**Review by RytoEX — APPROVED** [review]

**Pull request merged**